pkg/controller/controllercmd/builder.go

```diff
@@ -51,11 +55,14 @@ func NewController(componentName string, startFunc StartFunc) *ControllerBuilder
 	return &ControllerBuilder{
 		startFunc:        startFunc,
 		componentName:    componentName,
+		eventRecorder:    events.NewLoggingEventRecorder(),
 		observerInterval: defaultObserverInterval,
 	}
 }
 
-func (b *ControllerBuilder) WithFileObserver(reactorFunc func(file string, action fileobserver.ActionType) error, files ...string) *ControllerBuilder {
+// WithRestartOnChange will enable a file observer controller loop that observes changes into specified files. If a change to a file is detected,
+// the process will be terminated with return code zero.
+func (b *ControllerBuilder) WithRestartOnChange(stopCh chan struct{}, files ...string) *ControllerBuilder {
```

Comment:
Godoc is outdated.

Make stopCh a `chan<- struct{}`.

Reply:
fixed